### PR TITLE
fix(k8s): make `replace` op after disruption not raise false errors

### PR DESCRIPTION
**Pull request by vponomaryov**
Running `disrupt_drain_kubernetes_node_then_replace_scylla_node` nemesis we get following false error event:

```
  2023-08-21 15:10:21.838: (DatabaseLogEvent Severity.ERROR) \
    period_type=one-time event_id=<foo-id>: type=DATABASE_ERROR \
    regex=(^ERROR|!\s*?ERR).*\[shard.*\] line_number=19827 \
    node=sct-cluster-2-us-east1-b-us-east1-0
  ERROR 2023-08-21 15:09:17,271 [shard 0] \
    init - Startup failed: seastar::sleep_aborted (Sleep is aborted)
```

It is expected because we `replace` a Scylla pod which is in the `init` phase after disruption.

So, just filter this error event out in this nemesis.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by vponomaryov** [comment]
Example of the false SCT error event: https://argus.scylladb.com/test/335a7ae9-cd96-41a4-be35-76d1bf1617ff/runs?additionalRuns[]=168a76e5-bb6d-4668-8ba2-ecf8a58e3179

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -1379,8 +1379,12 @@ def _disrupt_kubernetes_then_replace_scylla_node(self, disruption_method):
 
         self.log.info('Mark %s (uid=%s) to be replaced', node, old_uid)
         node.wait_for_svc()
-        node.mark_to_be_replaced()
-        self._kubernetes_wait_till_node_up_after_been_recreated(node, old_uid=old_uid)
+        with DbEventsFilter(
+                db_event=DatabaseLogEvent.DATABASE_ERROR,
+                line="init - Startup failed.*seastar.*sleep_aborted"):
```
Comment (soyacz) [inline sdcm/nemesis.py:1384]: I'm not sure `line` accepts regex.
Why not to use just `init - Startup failed: seastar::sleep_aborted` ?

If you use regex, need to add .* at the beginning and the end.

Comment (vponomaryov) [reply sdcm/nemesis.py]: Done

**Review by soyacz — APPROVED** [review]
lgtm

**Review by juliayakovlev — APPROVED** [review]

**Pull request merged**